mono/mini/interp/transform.c

```diff
@@ -1166,13 +1166,23 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 	}
 
 	if (csignature->call_convention == MONO_CALL_VARARG) {
-		char *fullname = mono_method_full_name (method, TRUE);
-		mono_error_set_execution_engine (error, "__arglist not supported yet: %s\n", fullname);
-		g_free (fullname);
-		return;
+		csignature = mono_method_get_signature_checked (target_method, image, token, generic_context, error);
+		/*
+		 * For vararg calls, ArgIterator expects the signature and the varargs to be
+		 * stored in a linear memory. We allocate the necessary vt_stack space for
+		 * this. All varargs will be pushed to the vt_stack at call site.
+		 */
+		vt_stack_used += sizeof (gpointer);
+		PUSH_VT (td, sizeof (gpointer));
+		for (i = csignature->sentinelpos; i < csignature->param_count; ++i) {
+			int align, arg_size;
+			arg_size = mono_type_stack_size (csignature->params [i], &align);
+			vt_stack_used += arg_size;
+			PUSH_VT (td, arg_size);
+		}
 	}
 
-	g_assert (csignature->call_convention == MONO_CALL_DEFAULT || csignature->call_convention == MONO_CALL_C || csignature->call_convention == MONO_CALL_STDCALL);
+	g_assert (csignature->call_convention == MONO_CALL_DEFAULT || csignature->call_convention == MONO_CALL_C || csignature->call_convention == MONO_CALL_STDCALL || csignature->call_convention == MONO_CALL_VARARG);
```

Comment:
I guess we can invert this assert now to `g_assert (csignature->call_convention != MONO_CALL_THISCALL && csignature->call_convention != MONO_CALL_FASTCALL)` 🙂